dimagi_lake/aggregation/agg_table_schema.py

```diff
@@ -112,3 +115,44 @@ def write_to_warehouse(self):
                     drop_table(cursor, prev_table)
         finally:
             conn.close()
+
+
+class FlwcAdministration(BaseTable):
+    schema = StructType(fields=[
+        StructField('domain', StringType(), True),
+        StructField('month', DateType(), True),
+        StructField('flwc_id', StringType(), True),
+        StructField('supervisor_id', StringType(), True),
+        StructField('project_id', StringType(), True),
+        StructField('district_id', StringType(), True),
+        StructField('state_id', StringType(), True),
+        StructField('clean_drinking_water', IntegerType(), True),
+        StructField('functional_toilet', IntegerType(), True),
+        StructField('weighing_scale_infants', IntegerType(), True),
+        StructField('weighing_scale_mother_and_child', IntegerType(), True),
+        StructField('medicine_kit_available', IntegerType(), True),
+        StructField('infantometer_available', IntegerType(), True),
+        StructField('stadiometer_available', IntegerType(), True),
+    ])
+    _aggregator = AggFlwcAdministartionHelper
+    _warehouse_base_table = FLWC_ADMINISTRATION_TABLE
+    _partition_columns = ('month',)
+
+    def aggregate(self):
+        aggregator = self._aggregator(self._database_name,
+                                      self._domain,
+                                      self._month,
+                                      self.schema)
+        agg_df = aggregator.aggregate()
+        agg_df.show()
+        self.write_to_datalake(agg_df)
+
+    def write_to_datalake(self, df):
```

Comment:
if `write_to_datalake` is overridden, what value is being provided by the base class?

would a config class to bundle domain, month, and DB access params that each of the aggregators can use make it possible to get rid of the inheritance pattern? or you think it's unusual that `write_to_datalake` would be overridden and this is an exception

Reply:
Actually, There are two kinds of writes we want to make to datalake.
1. To complete overwrite of table like in location table.
2. Overwrite for only a selected month like other aggregation tables. 

I have fixed the inheritance in the other Prs. Missed here.
Fixing now. Basically, I think monthly partition overwrite would be more common, and completely overwrite would be rare. But would probably keep both options in the parent class.